acoustid/future/fingerprint.py

```
@@ -0,0 +1,56 @@
+    The function decompresses the zstd data, unpacks the bytes into integers,
+    and reconstructs the original hash values by reversing the XOR differences.
+    """
+
+    try:
+        data = zstd.decompress(data)
+    except zstd.Error as e:
+        raise ValueError(f"Failed to decompress fingerprint: {e}") from e
+
+    magic, fmt, version, *diffs = struct.unpack(f"<HBB{len(data) // 4 - 1}i", data)
+    if magic != MAGIC:
+        raise ValueError(f"Invalid fingerprint magic: {magic:#x}, expected: {MAGIC:#x}")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)